### ISSUE TITLE
Timeout while using objection 

### ISSUE THREAD

**Issue by sushi2k** [sensepost/objection#21]
Hi,

we just repackaged an IPA and installed it via ios-deploy:

```
ios-deploy --bundle Payload/XXX.app -W -d
```

The app is starting on the non-jailbroken phone and objection is also identifying the app 

```
➜  ~ frids-ps -U
zsh: command not found: frids-ps
➜  ~ frida-ps -U
PID  Name
---  ------
467  Gadget
➜  ~ objection explore

     _     _         _   _
 ___| |_  |_|___ ___| |_|_|___ ___
| . | . | | | -_|  _|  _| | . |   |
|___|___|_| |___|___|_| |_|___|_|_|
        |___|(object)inject(ion) v1.1.4

     Runtime Mobile Exploration
        by: @leonjza from @sensepost

[tab] for command suggestions
com.empty.awesome on (iPhone: 10.3.1) [usb] # env

Name                           Path
-----------------------------  ------------------------------------------------------------------------------------------------------------
DocumentDirectory              /var/mobile/Containers/Data/Application/1A1714A9-8F70-49BF-9ED0-AA400DD34391/Documents
LibraryDirectory               /var/mobile/Containers/Data/Application/1A1714A9-8F70-49BF-9ED0-AA400DD34391/Library
CachesDirectory                /var/mobile/Containers/Data/Application/1A1714A9-8F70-49BF-9ED0-AA400DD34391/Library/Caches
...
```

But if I want to navigate or make a 'ls' I get a timeout. Any idea why this is happening?

```bash
com.empty.awesome on (iPhone: 10.3.1) [usb] # cd /var/mobile/Containers/Data/Application/1A1714A9-8F70-49BF-9ED0-AA400DD34391/Documents
Exception in thread Thread-9:
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/sven/Library/Python/3.6/lib/python/site-packages/prompt_toolkit/interface.py", line 865, in run
    completions = list(buffer.completer.get_completions(document, complete_event))
  File "/usr/local/lib/python3.6/site-packages/objection/console/completer.py", line 103, in get_completions
    commands.update(self.find_completions(document))
  File "/usr/local/lib/python3.6/site-packages/objection/console/completer.py", line 65, in find_completions
    current_suggestions = current_suggestions[candidate]['dynamic']()
  File "/usr/local/lib/python3.6/site-packages/objection/commands/filemanager.py", line 852, in list_folders_in_current_fm_directory
    response = _get_short_ios_listing()
  File "/usr/local/lib/python3.6/site-packages/objection/commands/filemanager.py", line 761, in _get_short_ios_listing
    runner.run()
  File "/usr/local/lib/python3.6/site-packages/objection/utils/frida_transport.py", line 327, in run
    script.load()
  File "/usr/local/lib/python3.6/site-packages/frida/core.py", line 300, in load
    self._impl.load()
frida.TransportError: timeout was reached

/var/mobile/Containers/Data/Application/1A1714A9-8F70-49BF-9ED0-AA400DD34391/Documents
```

**Comment by leonjza** [comment]
Thanks for the report! My first thought is that the Frida hook used to get directory listings is taking longer than it should, maybe because of the number of files in the target directory. Having said that, I am guessing this is an 'empty' app (given the name) so that might not be the case.

As for the stack trace itself, the timeout is occurring where the REPL is waiting for the `ls` hook to finish to populate TAB completions for files and directories. To combat the timeout case, I am going to move some of the file manager internals to make use of the Frida RPC exports feature. This will at least allow for some more time for it to complete.

Is it possible to share this IPA with me to help with debugging?

**Comment by leonjza** [comment]
Does running `ls /` to list what is in the devices root directory also timeout?

**Comment by leonjza** [comment]
Just released v1.1.5 that changes how the Frida scripts for iOS file listings is used. Could you update and check if you still have timeouts?

Thanks!

**Comment by leonjza** [comment]
Closing this for now, if there are any other problems, let me know!
